modules/xfem/include/userobjects/LevelSetCutUserObject.h

```diff
@@ -0,0 +1,48 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#ifndef LEVELSETCUTUSEROBJECT_H
+#define LEVELSETCUTUSEROBJECT_H
+
+#include "GeometricCutUserObject.h"
+
+class LevelSetCutUserObject;
+
+template <>
+InputParameters validParams<LevelSetCutUserObject>();
+
+class LevelSetCutUserObject : public GeometricCutUserObject
+{
+public:
+  LevelSetCutUserObject(const InputParameters & parameters);
+
+  virtual bool cutElementByGeometry(const Elem * elem,
+                                    std::vector<Xfem::CutEdge> & cut_edges,
+                                    std::vector<Xfem::CutNode> & cut_nodes,
+                                    Real time) const override;
+  virtual bool cutElementByGeometry(const Elem * elem,
+                                    std::vector<Xfem::CutFace> & cut_faces,
+                                    Real time) const override;
+
+  virtual bool cutFragmentByGeometry(std::vector<std::vector<Point>> & frag_edges,
+                                     std::vector<Xfem::CutEdge> & cut_edges,
+                                     Real time) const override;
+  virtual bool cutFragmentByGeometry(std::vector<std::vector<Point>> & frag_faces,
+                                     std::vector<Xfem::CutFace> & cut_faces,
+                                     Real time) const override;
+
+  virtual const std::vector<Point>
+  getCrackFrontPoints(unsigned int num_crack_front_points) const override;
+
+protected:
+  /// The variable number of the level set variable we are operating on
+  const unsigned int _ls_var_number;
+
+  const System & _system;
+  const NumericVector<Number> * _solution;
```

Comment:
No, they are general libmesh(?) system and solution. They are here because I need to find the level set value at the node rather than quadrature points.

Reply:
Thanks for the explanation! Would you add this information as a doxygen comment for the two variables please?